include/kodlab_mjbots_sdk/rotations.h

```diff
@@ -0,0 +1,410 @@
+
+  Eigen::Matrix3<Scalar> m;
+  m.coeffRef(0, 0) = c2 * c1;
+  m.coeffRef(0, 1) = c2_s1 * s0 - s2 * c0;
+  m.coeffRef(0, 2) = c2_s1 * c0 + s2 * s0;
+  m.coeffRef(1, 0) = s2 * c1;
+  m.coeffRef(1, 1) = s2_s1 * s0 + c2 * c0;
+  m.coeffRef(1, 2) = s2_s1 * c0 - c2 * s0;
+  m.coeffRef(2, 0) = -s1;
+  m.coeffRef(2, 1) = c1 * s0;
+  m.coeffRef(2, 2) = c1 * c0;
+
+  return m;
+}
+
+/**
+ * @brief Constructs a 3x3 rotation matrix from a normalized quaternion.
+ * @note See the cited Drake code for details
+ * @cite Russ Tedrake and the Drake Development Team, "Drake: Model-based design
+ *       and verification for robotics" (2019)
+ * @link https://github.com/RobotLocomotion/drake/blob/5022e3832e90f988f1b6d3015b2794ea26e527b1/math/rotation_matrix.h#L970 @endlink
+ *
+ * @tparam Scalar[optional] numeric type (default float)
+ * @param quaternion
+ * @param two_over_norm_squared
+ * @return Eigen::Matrix3<Scalar>
+ */
+template<typename Scalar = float>
+inline Eigen::Matrix3<Scalar> QuaternionToRotationMatrix(
+    const Eigen::Quaternion<Scalar> &quaternion,
+    const Scalar &two_over_norm_squared)
+{
+  const Scalar w = quaternion.w();
+  const Scalar x = quaternion.x();
+  const Scalar y = quaternion.y();
+  const Scalar z = quaternion.z();
+  const Scalar sx = two_over_norm_squared * x; // scaled x-value.
+  const Scalar sy = two_over_norm_squared * y; // scaled y-value.
+  const Scalar sz = two_over_norm_squared * z; // scaled z-value.
+  const Scalar swx = sx * w;
+  const Scalar swy = sy * w;
+  const Scalar swz = sz * w;
+  const Scalar sxx = sx * x;
+  const Scalar sxy = sy * x;
+  const Scalar sxz = sz * x;
+  const Scalar syy = sy * y;
+  const Scalar syz = sz * y;
+  const Scalar szz = sz * z;
+
+  Eigen::Matrix3<Scalar> m;
+  m.coeffRef(0, 0) = Scalar(1) - syy - szz;
+  m.coeffRef(0, 1) = sxy - swz;
+  m.coeffRef(0, 2) = sxz + swy;
+  m.coeffRef(1, 0) = sxy + swz;
+  m.coeffRef(1, 1) = Scalar(1) - sxx - szz;
+  m.coeffRef(1, 2) = syz - swx;
+  m.coeffRef(2, 0) = sxz - swy;
+  m.coeffRef(2, 1) = syz + swx;
+  m.coeffRef(2, 2) = Scalar(1) - sxx - syy;
+
+  return m;
+}
+
+/**
+ * @brief Constructs a 3x3 rotation matrix from a normalized quaternion.
+ * @note See the cited Drake code for details
+ * @cite Russ Tedrake and the Drake Development Team, "Drake: Model-based design
+ *       and verification for robotics" (2019)
+ * @link https://github.com/RobotLocomotion/drake/blob/5022e3832e90f988f1b6d3015b2794ea26e527b1/math/rotation_matrix.h#L970 @endlink
+ *
+ * @tparam Scalar[optional] numeric type (default float)
+ * @param unit_quaternion
+ * @return Eigen::Matrix3<Scalar>
+ */
+template<typename Scalar = float>
+inline Eigen::Matrix3<Scalar> QuaternionToRotationMatrix(
+    const Eigen::Quaternion<Scalar> &unit_quaternion)
+{
+  return QuaternionToRotationMatrix(unit_quaternion, Scalar(2));
+}
+
+/**
+ * @brief Constructs an euler angle representation from a normalized quaternion
+ *        and a rotation matrix
```

Comment:
should mention that this is specific to roll pitch yaw (and which RPY)

Reply:
Clarified in bdfce1322a036db5ad1a8e1a939c31b6042e2025